pkg/apis/componentconfig/types.go

```diff
@@ -176,6 +176,14 @@ const (
 type KubeletConfiguration struct {
 	metav1.TypeMeta
 
+	// Only used for dynamic configuration.
+	// The length of the trial period for this configuration. Default 10 minutes.
+	ConfigTrialDuration metav1.Duration
+	// Only used for dynamic configuration.
+	// The maximum number of Kubelet restarts that can occur in ConfigTrialDuration before the
+	// configuration fails the trial and the Kubelet rolls back to its last-known-good config.
+	// Default 3, mimimum allowed is 2, maximum allowed is 10.
```

Comment:
Ah, thanks for that catch on the default vs. the comment.

The Kubelet has to restart at least once to take up the new configuration, and since the curSymlink would have been updated when the new configuration was downloaded, the startups file will contain one restart since that update. So we can't count the restart as a "crash" until we see 2 restarts.